src/Testcontainers/Clients/TestcontainersClient.cs

```diff
@@ -327,9 +319,22 @@ await Task.WhenAll(configuration.ResourceMappings.Values.Select(resourceMapping
     /// <inheritdoc />
     public async Task<string> BuildAsync(IImageFromDockerfileConfiguration configuration, CancellationToken ct = default)
     {
+      var dockerfileFilePath = Path.Combine(configuration.DockerfileDirectory, configuration.Dockerfile);
+
       var cachedImage = await Image.ByNameAsync(configuration.Image.FullName, ct)
         .ConfigureAwait(false);
 
+      if (File.Exists(dockerfileFilePath))
+      {
+        await Task.WhenAll(File.ReadAllLines(dockerfileFilePath)
+          .Select(line => FromLinePattern.Match(line))
+          .Where(match => match.Success)
+          .Select(match => match.Groups["image"])
+          .Select(group => group.Value)
+          .Select(image => new DockerImage(image))
```

Comment:
I didn't see the constructor for the whole image string...  This way the whole thing looks much cleaner and you can save all the code I wrote for extrating repository, image and tag. 👍